packages/lodestar/src/chain/chain.ts

```diff
@@ -193,6 +194,70 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     };
   }
 
+  /**
+   * Restore state cache and forkchoice from last finalized state.
+   */
+  private async restoreFromFinalized(): Promise<void> {
+    const start = Date.now();
+    const lastFinalizedState: TreeBacked<BeaconState> = await this.db.stateArchive.lastValue();
+    this.db.stateCache.add(lastFinalizedState);
+    // the block respective to finalized epoch is still in block db
+    const allBlocks = await this.db.block.values();
```

Comment:
Aren't you suppose to search blocks from lastFinalizedState.slot?

Reply:
I think in the end we'll load all blocks in order to build forkchoice & stateCache so I load all from the beginning, did not compare the performance against loading block one by one from db on demand

Reply:
Makes sense, non finalized blocks are in archive anyways. We can leave this for now, we should probably store even some non finalized state in case of long non finalization like currently on schlesi.